### ISSUE TITLE
google.api_core.exceptions.ServiceUnavailable: 503 Getting metadata from plugin failed with error: None could not be converted to unicode

### ISSUE THREAD

**Issue by j-adamczyk** [googleapis/python-language#402]
#### Environment details

  - OS type and version: Ubuntu 20.04
  - Python version: Python 3.9.14
  - pip version: pip 22.2.2
  - `google-cloud-language` version: 
```
Name: google-cloud-language
Version: 2.6.1
Summary: Google Cloud Natural Language API client library
Home-page: https://github.com/googleapis/python-language
Author: Google LLC
Author-email: googleapis-packages@google.com
License: Apache 2.0
Location: /home/jakub/.cache/pypoetry/virtualenvs/predict-api-python-58y2ik0X-py3.9/lib/python3.9/site-packages
Requires: google-api-core, proto-plus, protobuf
Required-by: 

```

#### Steps to reproduce

Run this simple async hello world example:
```
import asyncio
import os

from google.cloud import language_v1
from google.cloud.language_v1 import Document, EncodingType

os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = "credentials.json"


async def sample_analyze_sentiment():
    # Create a client
    client = language_v1.LanguageServiceAsyncClient()

    texts = ["Good morning", "Hello there", "General Kenobi"]

    documents = [
        {
            "content": text,
            "type_": Document.Type.PLAIN_TEXT,
            "language": "en",
        }
        for text in texts
    ]

    requests = [
        {"document": document, "encoding_type": EncodingType.UTF8}
        for document in documents
    ]

    tasks = [client.analyze_sentiment(request) for request in requests]

    response = await gather(*tasks)

    # Handle the response
    print(response)


asyncio.run(sample_analyze_sentiment())
```

I get errors about 80% of time, but sometimes (randomly) it works and I get result:
```
[document_sentiment {
  magnitude: 0.8
  score: 0.8
}
language: "en"
sentences {
  text {
    content: "Good morning"
  }
  sentiment {
    magnitude: 0.8
    score: 0.8
  }
}
, document_sentiment {
  magnitude: 0.3
  score: 0.3
}
language: "en"
sentences {
  text {
    content: "Hello there"
  }
  sentiment {
    magnitude: 0.3
    score: 0.3
  }
}
, document_sentiment {
  magnitude: 0.1
  score: 0.1
}
language: "en"
sentences {
  text {
    content: "General Kenobi"
  }
  sentiment {
    magnitude: 0.1
    score: 0.1
  }
}
]
```

It always works when I use a single element await, instead of gather:
```
# tasks = [client.analyze_sentiment(request) for request in requests]
response = await client.analyze_sentiment(requests[0])
```

#### Stack trace
```
Traceback (most recent call last):
  File "/home/jakub/.cache/pypoetry/virtualenvs/predict-api-python-58y2ik0X-py3.9/lib/python3.9/site-packages/google/api_core/grpc_helpers_async.py", line 81, in __await__
    response = yield from self._call.__await__()
  File "/home/jakub/.cache/pypoetry/virtualenvs/predict-api-python-58y2ik0X-py3.9/lib/python3.9/site-packages/grpc/aio/_call.py", line 290, in __await__
    raise _create_rpc_error(self._cython_call._initial_metadata,
grpc.aio._call.AioRpcError: <AioRpcError of RPC that terminated with:
	status = StatusCode.UNAVAILABLE
	details = "Getting metadata from plugin failed with error: None could not be converted to unicode"
	debug_error_string = "UNKNOWN:Error received from peer language.googleapis.com:443 {created_time:"2022-11-15T13:52:42.29405481+01:00", grpc_status:14, grpc_message:"Getting metadata from plugin failed with error: None could not be converted to unicode"}"
>

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/jakub/PycharmProjects/predict_api_python/tmp.py", line 38, in <module>
    asyncio.run(sample_analyze_sentiment())
  File "/usr/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/home/jakub/PycharmProjects/predict_api_python/tmp.py", line 32, in sample_analyze_sentiment
    response = await asyncio.gather(*tasks)
  File "/home/jakub/.cache/pypoetry/virtualenvs/predict-api-python-58y2ik0X-py3.9/lib/python3.9/site-packages/google/cloud/language_v1/services/language_service/async_client.py", line 312, in analyze_sentiment
    response = await rpc(
  File "/home/jakub/.cache/pypoetry/virtualenvs/predict-api-python-58y2ik0X-py3.9/lib/python3.9/site-packages/google/api_core/grpc_helpers_async.py", line 84, in __await__
    raise exceptions.from_grpc_error(rpc_error) from rpc_error
google.api_core.exceptions.ServiceUnavailable: 503 Getting metadata from plugin failed with error: None could not be converted to unicode
```

**Comment by j-adamczyk** [comment]
Also, using synchronous API, even in parallel threads, works fine:
```
client = LanguageServiceClient()
... # code same as above
analyze = lambda request: client.analyze_sentiment(request)
tasks = [asyncio.to_thread(analyze, request=request) for request in requests]

response = await asyncio.gather(*tasks)

# Handle the response
print(response)
```

**Comment by parthea** [comment]
I ran the async sample 10 times but was not able to re-produce this issue. It's possible that the issue was fixed on the backend since 5XX errors are typically backend related. I'm going to close this issue but please open a new issue if the problem re-appears.
